lib/spack/spack/cmd/style.py

```diff
@@ -32,39 +34,49 @@ def grouper(iterable, n, fillvalue=None):
     "Collect data into fixed-length chunks or blocks"
     # grouper('ABCDEFG', 3, 'x') --> ABC DEF Gxx"
     args = [iter(iterable)] * n
-    return zip_longest(*args, fillvalue=fillvalue)
+    for group in zip_longest(*args, fillvalue=fillvalue):
+        yield filter(None, group)
+
 
+#: directory where spack style started, for relativizing paths
+initial_working_dir = None
 
 #: List of directories to exclude from checks.
 exclude_directories = [spack.paths.external_path]
 
-#: max line length we're enforcing (note: this duplicates what's in .flake8)
-max_line_length = 79
+#: order in which tools should be run. flake8 is last so that it can
+#: double-check the results of other tools (if, e.g., --fix was provided)
+tool_order = ["isort", "mypy", "black", "flake8"]
 
+#: tools we run in spack style
+tools = {}
 
-def is_package(f):
-    """Whether flake8 should consider a file as a core file or a package.
 
-    We run flake8 with different exceptions for the core and for
-    packages, since we allow `from spack import *` and poking globals
-    into packages.
-    """
-    return f.startswith("var/spack/repos/") or "docs/tutorial/examples" in f
+#: decorator for adding tools to the list
+class tool(object):
+    def __init__(self, name, required=False):
+        self.name = name
+        self.required = required
+
+    def __call__(self, fun):
+        tools[self.name] = (fun, self.required)
+        return fun
 
 
 def changed_files(base=None, untracked=True, all_files=False):
     """Get list of changed files in the Spack repository."""
 
     git = which("git", required=True)
 
+    # GITHUB_BASE_REF is set to the base branch for pull request actions
     if base is None:
-        base = os.environ.get("TRAVIS_BRANCH", "develop")
+        base = os.environ.get("GITHUB_BASE_REF", "develop")
 
     range = "{0}...".format(base)
 
     git_args = [
         # Add changed files committed since branching off of develop
-        ["diff", "--name-only", "--diff-filter=ACMR", range],
+        ["diff", "--name-only", "--diff-filter=ACMR", "--", range],
```

Comment:
Reading the docs it seems that `--` is used to distinguish commit ref from paths,
```suggestion
        ["diff", "--name-only", "--diff-filter=ACMR", range],
```
so I think it should be omitted here?

Reply:
Nice catch. I had to do this to get things to work for some reason but I cannot reproduce now.  Deleted.